apps/minifront/src/state/shared.ts

```diff
@@ -16,10 +22,70 @@ export const { stakingTokenMetadata, useStakingTokenMetadata } = createZQuery({
   },
 });
 
+export const { balancesResponses, useBalancesResponses } = createZQuery({
+  name: 'balancesResponses',
+  fetch: getBalances,
+  getUseStore: () => useStore,
+  get: state => state.shared.balancesResponses,
+  set: setter => {
+    const newState = setter(useStore.getState().shared.balancesResponses);
+    useStore.setState(state => {
+      state.shared.balancesResponses = newState;
+    });
+  },
+});
+
+export const { assets, useAssets } = createZQuery({
+  name: 'assets',
+  fetch: getAllAssets,
+  getUseStore: () => useStore,
+  get: state => state.shared.assets,
+  set: setter => {
+    const newState = setter(useStore.getState().shared.assets);
+    useStore.setState(state => {
+      state.shared.assets = newState;
+    });
+  },
+});
+
 export interface SharedSlice {
+  assets: ZQueryState<Metadata[]>;
+  balancesResponses: ZQueryState<BalancesResponse[]>;
   stakingTokenMetadata: ZQueryState<Metadata>;
 }
 
 export const createSharedSlice = (): SliceCreator<SharedSlice> => () => ({
+  assets,
+  balancesResponses,
   stakingTokenMetadata,
 });
+
+export interface BalancesByAccount {
+  account: number;
+  address: Address;
+  balances: BalancesResponse[];
+}
+
+export const groupByAccount = (
+  acc: BalancesByAccount[],
+  curr: BalancesResponse,
+): BalancesByAccount[] => {
+  const index = getAddressIndex(curr.accountAddress);
+  const grouping = acc.find(a => a.account === index.account);
+
+  if (grouping) {
+    grouping.balances.push(curr);
+  } else {
+    acc.push({
+      account: index.account,
+      address: getAddress(curr.accountAddress),
+      balances: [curr],
+    });
+  }
+
+  return acc;
+};
+
+export const balancesByAccountSelector = (
```

Comment:
Normally, I'd colocate a use case-specific selector with the component that uses it. But this is used by multiple components, so I figured it belongs colocated with the source of the data.